services/transactionpool/service.go

```diff
@@ -27,12 +27,18 @@ func NewTransactionPool(gossip gossiptopics.TransactionRelay, reporting instrume
 }
 
 func (s *service) AddNewTransaction(input *services.AddNewTransactionInput) (*services.AddNewTransactionOutput, error) {
-	s.reporting.Info(fmt.Sprintf("Adding new transaction [%v] to the pool", input.SignedTransaction))
-	s.gossip.BroadcastForwardedTransactions(&gossiptopics.ForwardedTransactionsInput{
-		Message: &gossipmessages.ForwardedTransactionsMessage{
-			SignedTransactions: []*protocol.SignedTransaction{input.SignedTransaction},
-		},
-	})
+	err := validateTransaction(input.SignedTransaction)
+	if err != nil {
+		s.reporting.Info(fmt.Sprintf("transaction is invalid [%v]", input.SignedTransaction))
+		return nil, err
+	} else {
```

Comment:
as much as I hate it, we decided to not use `else` after `return`, so we'd better stick to the style

Reply:
Kiss to @electricmonk !